### DISCUSSION TITLE
Unable to create a simple tree structure in data

### DISCUSSION THREAD

**Discussion by leolivier** [Budibase/budibase#12685] [Ideas]
Hi,
I need to create a very basic tree structure in a table. Let's say I have a table named A, I want to add a "parent" relationship from A to A.
The bb data UI prevents me in doing this: A is not listed in the relationship list of tables...

**Comment by melohagan** [comment]
Hey @leolivier 

There is an open feature request for defining relationships within the same table: https://github.com/Budibase/budibase/issues/3498

Possibly you could use custom queries to achieve what you are looking to do: https://docs.budibase.com/docs/data-sources#custom-queries

**Reply by leolivier** [reply]
Unfortunately, I am using budibase db which doesn't support that 😬

**Reply by melohagan** [reply]
Hmm, could you add a number column that references an ID column, and then filter on that column?

E.g.

| ID | ParentID | 
| -- | -- |
| 1 | - |
| 2 | 1 |
|3 | 1 |

**Reply by leolivier** [reply]
Yeah, that's what I did for now (using a name to make it more user friendly) but when creating a new A, I can't be sure the value provided in the parentID is really an A id.
I had to create option picker containing the possible values of parent but this isn't possible in a side panel so I had to create a specific form :/
